docs/design/design_v1.8.md

```diff
@@ -0,0 +1,437 @@
+
+kubeadm relies on the kubelet to pull the control plane images and run them properly as Static Pods.
+But there are (as we've seen) a lot of things that can go wrong. Most of them are network/resolv.conf/proxy related.
+
+After the control plane is up, kubeadm completes a couple of tasks described in following paragraphs.
+
+### Saves MasterConfiguration in a ConfigMap for later reference
+
+kubeadm saves the configuration passed to `kubeadm init`, either via flags or the config file,  in a ConfigMap named `kubeadm-config ` under `kube-system` namespace.
+
+This will ensure that kubeadm actions executed in future (e.g `kubeadm upgrade`) will be able to determine the actual/current cluster state and make new decisions based on that data.
+
+Please note that 
+
+1. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+2. Upload of master configuration can be invoked individually with the `kubeadm phase upload-config` command.
+3. If you initialized your cluster using kubeadm v1.7.x or lower, you must create manually the master configuration ConfigMap before `kubeadm upgrade` to v1.8 . In order to facilitate this task, the ` kubeadm config upload (from-flags|from-file)` was implemented.
+
+### Mark master
+
+As soon as the control plane is available, kubeadm executes following actions: 
+
+- Label  the master with `node-role.kubernetes.io/master=""` 
+- Taints  the master with `node-role.kubernetes.io/master:NoSchedule`
+
+Please note that 
+
+1. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+2. Mark master phase can be invoked individually with the `kubeadm phase mark-master` command.
+
+### Configure TLS-Bootstrapping for node joining
+
+Kubeadm uses [Authenticating with Bootstrap Tokens](https://kubernetes.io/docs/admin/bootstrap-tokens/) for joining new nodes to an existing cluster.
+
+`kubeadm init`  ensures that everything is properly configured for this process, and this includes following steps as well as setting API Server and controller flags as already described in previous paragraphs.
+
+#### Save token in a secret for later reference
+
+`kubeadm init`  create a first bootstrap token, either generated automatically or provided by the user with the `--token` flag; as documented in bootstrap token specification, token should be saved as secrets with name `bootstrap-token-<token-id>` under `kube-system` namespace.
+
+Please note that 
+
+1. The token will be used to validate temporary user during TLS bootstrap process; those users will be member of  `system:bootstrappers:kubeadm:default-node-token` group (formerly `system:bootstrappers` in v1.7)
+2. Starting from 1.8 token has a limited validity, default 24Hours (that can be changed with `—token-ttl` flag)
+3. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+4. Additional tokens can be created/managed with the `kubeadm token` command.
```

Comment:
Please move this out of the list to above and write in a command block with the `kubeadm token` command `kubeadm init` would perform